signal-hook-tokio/src/lib.rs

```diff
@@ -73,178 +117,42 @@ macro_rules! implement_signals_with_pipe {
     };
 }
 
-/// A module for integrating signal handling with the Tokio 0.1 runtime.
-///
-/// This provides the [`Signals`][v0_1::Signals] struct which acts as a
-/// [`Stream`][`tokio_0_1::prelude::Stream`] of signals.
-///
-/// # Example
-///
-/// ```rust
-/// # use tokio_0_1 as tokio;
-/// use std::io::Error;
-///
-/// use signal_hook::consts::signal::*;
-/// use signal_hook_tokio::v0_1::Signals;
-/// use tokio::prelude::*;
-///
-/// enum SignalResult {
-///     Err(Error),
-///     Shutdown,
-/// }
-///
-/// fn main() -> Result<(), Error> {
-///     let signals = Signals::new(&[
-///             SIGTERM,
-/// #           SIGUSR1,
-///         ])?
-///         .map_err(|error| {
-///             SignalResult::Err(error)
-///         })
-///         .for_each(|sig| {
-///             // Return an error to stop the for_each iteration.
-///             match sig {
-///                 SIGTERM => return future::err(SignalResult::Shutdown),
-/// #               SIGUSR1 => return future::err(SignalResult::Shutdown),
-///                 _ => unreachable!(),
-///             }
-///         })
-///         .map_err(|reason| {
-///             if let SignalResult::Err(error) = reason {
-///                 eprintln!("Internal error: {}", error);
-///             }
-///         });
-///
-/// #   signal_hook::low_level::raise(SIGUSR1).unwrap();
-///     tokio::run(signals);
-///     Ok(())
-/// }
-/// ```
-#[cfg(feature = "support-v0_1")]
-pub mod v0_1 {
-    use futures_0_1::stream::Stream;
-    use futures_0_1::{Async, Poll};
-    use signal_hook::iterator::backend::PollResult;
-    use tokio_0_1::io::AsyncRead;
-    use tokio_0_1::net::unix::UnixStream;
-
-    implement_signals_with_pipe!(UnixStream);
-
-    impl<E: Exfiltrator> SignalsInfo<E> {
-        /// Check if signals arrived by polling the stream for some bytes.
-        ///
-        /// Returns true if it was possible to read a byte and false otherwise.
-        fn has_signals(read: &mut UnixStream) -> Result<bool, Error> {
-            match read.poll_read(&mut [0u8]) {
-                Poll::Ok(Async::NotReady) => Ok(false),
-                Poll::Ok(Async::Ready(num_read)) => Ok(num_read > 0),
-                Poll::Err(error) => Err(error),
-            }
-        }
-    }
+use tokio::net::UnixStream;
+implement_signals_with_pipe!(UnixStream);
+
+cfg_if::cfg_if! {
+    if #[cfg(feature = "futures-v0_3")] {
```

Comment:
Do you think it would be possible not to pull in the extra dependency? It seems our situation is not that complex to really need it. I was thinking about just putting the `#[cfg(...)]` on the `impl Stream for Signals` block. I'd probably say it's better to just suppress unused functions warnings instead with `#[cfg_attr(not(feature = "...", allow(dead_code))]`. I mean, the crate will be unusable without the feature anyway, so being clean at that point is not that important.

Reply:
Conditionally suppressing the warnings is a great alternative that had not occurred to me. Thanks for the suggestion.